ipatests/test_integration/test_sudo.py

```diff
@@ -756,3 +757,1299 @@ def test_domain_resolution_order(self):
+        master = self.master
+        kinit_admin(master)
+
+        result = master.run_command([
+            "ipa", "sudorule-add-runasgroup", self.SUDO_RULE,
+            "--groups=ALL"
+        ], raiseonerr=False)
+        assert (
+            "ERROR: invalid 'runas-group': RunAsGroup does not accept 'ALL' "
+            "as a group name"
+        ) in result.stderr_text
+
+    # ----------------------
+    # Sudo disable/enable
+    # ----------------------
+    def test_sudorule_disable_func001(self):
+        master = self.master
+        kinit_admin(master)
+
+        result = master.run_command(["ipa", "sudorule-show", self.SUDO_RULE])
+        master.run_command(["ipa", "sudorule-disable", self.SUDO_RULE])
+        master.run_command(["ipa", "sudocmd-add", "/bin/newcmd"])
+        master.run_command([
+            "ipa", "sudorule-add-allow-command", "--sudocmds=/bin/newcmd",
+            self.SUDO_RULE
+        ])
+        clear_sssd_cache(master)
+        clear_sssd_cache(self.client)
+        result = self.list_sudo_commands(self.USER_1)
+        assert not re.search(r"\(root\).*?/bin/newcmd", result.stdout_text)
+
+        master.run_command(["ipa", "sudorule-enable", self.SUDO_RULE])
+        clear_sssd_cache(self.client)
+        result = self.list_sudo_commands(self.USER_1)
+        assert (
+            "(root) /bin/uname, /bin/touch, "
+            "/bin/date, /bin/newcmd, /bin/mkdir"
+            in result.stdout_text
+        )
+
+    # ----------------------
+    # Cleanup
+    # ----------------------
+    def test_cleanup(self):
+        master = self.master
+        kinit_admin(master)
+
+        # Delete sudo command group
+        master.run_command(["ipa", "sudocmdgroup-del", self.SUDO_GROUP],
+                           raiseonerr=False)
+        # Delete sudo rule
+        master.run_command(["ipa", "sudorule-del", self.SUDO_RULE],
+                           raiseonerr=False)
+        # Delete sudo commands
+        sudo_commands = [
+            "/bin/mkdir", "/bin/date", "/bin/df", "/bin/touch", "/bin/rm",
+            "/bin/uname", "/bin/hostname", "/bin/rmdir", "/bin/newcmd"
+        ]
+        for cmd in sudo_commands:
+            master.run_command(["ipa", "sudocmd-del", cmd], raiseonerr=False)
+        clear_sssd_cache(self.master)
+
+    # ----------------------
+    # Offline client caching functional test
+    # ----------------------
+    def test_001_sudorule_offline_caching_allow_command(self):
+        master = self.master
+        client = self.clients[0].hostname
+        kinit_admin(master)
+
+        self.setup_sudo_rule(
+            master=master,
+            client=client,
+            user=self.USER_1,
+            rule_name=self.SUDO_RULE,
+            group_name=self.SUDO_GROUP,
+            allowed_command="/bin/date"
+        )
+        clear_sssd_cache(master)
+        clear_sssd_cache(self.client)
+        result = self.run_as_sudo_user(
```

Comment:
The test doesn't check the result, either use the value in an assertion or replace with `self.run_as_sudo_user(`